Table tests

maelle opened the pull request:
To avoid conflicts with #685, experimenting here.

Cc @mpadge

I'm copying what happens for ropensci.org/packages, with [Row details](https://datatables.net/examples/server_side/row_details.html) to display more information. I first thought about adding some sort of card on hover, but I prefer this solution.

Rows are expanded when clicking on the closed book. Most citations have a clickable link "built-in" by @sckott but some don't so I use the DOI column to add a link shown by a link emoji. I only kept citations with a DOI (therefore excluding blog posts).

Several questions

* Is it actually a good idea to use JS to style the table? I did it to make it easier to interact with (searching, expanding fields) but then I wonder about folks blocking scripts, reading on mobile, etc.
* If it's a good idea, how can we improve the design?
* If it's not a good idea, how do we present the table?
* In all cases, do we want to show more information? On the packages page we have peer-review links, whether the package is on CRAN (also its status but I think we're only showing active packages in the blog post anyway).

maelle commented on content/blog/2020-04-30-table/index.Rmd:
````diff
@@ -0,0 +1,132 @@
+
+
+```{r table, echo=FALSE, message=FALSE}
+packages <- readr::read_tsv("data-access-pkgs.tsv")
````
I haven't added these two TSV in the PR, that's the Google sheet and [the citations file](https://github.com/ropenscilabs/ropensci_citations/blob/master/citations_all.tsv).

sckott commented:
- Some of the older citations i think are more likely to not have links. Correct, can easily make a link if there is a DOI by doing https://doi.org/{DOI}
- I really like the JS enable table. sure there will be some that have JS disabled - but we can point people to some links or so if they have it disabled
- showing more information: I'd lean towards not showing more information/columns since we're already showing a lot of information

stefaniebutland commented:
As I said on Slack, the content this table provides is amazing.

> In all cases, do we want to show more information? 

We don't want to duplicate the packages page (you brought this up in the mtg and I woke up Thurs am worried about this :-)) and it fits well in page now. 💡right above this table should say something like "Text to describe this table. For a full list of rOpenSci packages including their peer-review status, see our [Packages](/packages/) page.

- above table say total number of packages in table (I see it at bottom. Can it be moved to top? If not, add in text)

- 📕 my first impression was this was only an emoji. Need a (visual?) way to get people to click

- is the 🔗 emoji redundant since the doi url is active?  https://doi.org/10.1093/molbev/msw004 🔗

stefaniebutland commented:
Question: for data sources like "NCBI, ENSEMBL, UniProt" each has a different link. Is it possible to have each of those linked to source?

maelle commented:
Yes, and easier for me if the info is stored in a nested JSON then @sckott @mpadge

maelle commented:
Thanks @sckott @stefaniebutland

I made a few tweaks. I'll keep working a bit on this branch before adding it to the other PR.

* I decided to add the peer-review link when available, because it sounded better than telling readers to go read another table, and because I think it might add legitimacy like the citations. Feel free to disagree, I can remove it.

* I added a small "Click for details" at the top of the column with emojis, and I added a small text before the table.

* I removed DOIs and DOI links from the citations so only my "link emoji link" gives the DOI now.

maelle commented:
Note to self: explore making more hidden fields available for search like we do for the packages page (e.g. keywords).

stefaniebutland commented:
Open Software Peer Review

- I would put `Open Software Peer Review.` below the citations. For audience for this post I think the citations carry more weight.

- I would replace "Open Software Peer Review" -> "This package has passed open software peer review.", for clarity for intended audience.

stefaniebutland commented:
> I removed DOIs and DOI links from the citations so only my "link emoji link" gives the DOI now.

Open question: Is keeping the DOI link visible and not using emoji more recognizable to audience who sees DOI links in journals so recognizes their value?

maelle commented:
@stefaniebutland Thank you! I implemented all changes except for the location of the peer-review mention because some packages have so many citations that the software peer-review would be hidden. Maybe we should find a compromise making peer-review status some emoji like on the packages page?

@mpadge I used your table, thank you!
* You'll see I renamed fields at the beginning of the chunk to match @sckott's Google sheet.
* In Scott's sheet there was data source vs link so I could make data source clickable. I'm not sure how feasible it is with your automatic workflow but if a nested JSON were a possibility it'd be great e.g.

```json
{
  "package": "blabla",
  "data_source": [
    {
      "name": "cool",
      "url": "url1"
    },
    {
      "name": "great",
      "url": "url2"
    }
  ]
} 
```

Otherwise is the goal to have the
* I'd remove ghrecipes :-)

maelle commented:
Now when searching the "keywords" and "citations" fields are available so you'll get one package if you search for "secretome" or "France" (because of citations). You'll also get packages if you search for "unconf".

maelle commented:
@mpadge @sckott @stefaniebutland candidate for review before I include it in the other PR (I won't do that until the data input format stops changing).

Mark's data has more packages, but now data source can be either name or URL, not both, and some rows have no "data source" (see e.g. MODIStsp). It's not a problem IMO, since the _description_ of such packages if fine, and searching for MODIS would get the row which is the most important thing.

maelle commented:
@mpadge what happened to cRegulome?

mpadge commented:
> @mpadge what happened to cRegulome?

it's still there, and seems intact to me

maelle commented:
> it's still there, and seems intact to me

it used to be truncated, sorry, no longer the case

maelle commented:
I'll let @mpadge comment on these remarks since they're about the data we're feeding the table. Just a vague idea on "should we be finding a way to populate Data Source field where it's empty e.g. ccafs? (guessing source of that info comes from pkg authors)", maybe we should somehow merge "description" and "data source" for such cases?

stefaniebutland commented:
> should we be finding a way to populate Data Source field where it's empty e.g. ccafs? (guessing source of that info comes from pkg authors)

We decided to leave as is. Won't manually edit. Maëlle will open issue in dev guide to encourage pkg authors to have data source info in their pkg

maelle commented:
https://github.com/ropensci/dev_guide/issues/260 🙂

maelle commented:
@jeroen I'll use this PR as is in the post tomorrow unless you tweak it before then.

mpadge commented:
> Should https://docs.ropensci.org/tradestatistics/ should be in the table?

Absolutely! Yes, yes, and more yes!

maelle commented:
I added it by hand at the moment but feel free to edit that 😁
